api/v1beta1/argocd_types.go

```diff
@@ -0,0 +1,1022 @@
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Initial Repositories'",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text","urn:alm:descriptor:com.tectonic.ui:advanced"}
+	InitialRepositories string `json:"initialRepositories,omitempty"`
+
+	// InitialSSHKnownHosts defines the SSH known hosts data upon creation of the cluster for connecting Git repositories via SSH.
+	InitialSSHKnownHosts SSHHostsSpec `json:"initialSSHKnownHosts,omitempty"`
+
+	// KustomizeBuildOptions is used to specify build options/parameters to use with `kustomize build`.
+	KustomizeBuildOptions string `json:"kustomizeBuildOptions,omitempty"`
+
+	// KustomizeVersions is a listing of configured versions of Kustomize to be made available within ArgoCD.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Kustomize Build Options'",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text","urn:alm:descriptor:com.tectonic.ui:advanced"}
+	KustomizeVersions []KustomizeVersionSpec `json:"kustomizeVersions,omitempty"`
+
+	// OIDCConfig is the OIDC configuration as an alternative to dex.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OIDC Config'",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text","urn:alm:descriptor:com.tectonic.ui:advanced"}
+	OIDCConfig string `json:"oidcConfig,omitempty"`
+
+	// Monitoring defines whether workload status monitoring configuration for this instance.
+	Monitoring ArgoCDMonitoringSpec `json:"monitoring,omitempty"`
+
+	// NodePlacement defines NodeSelectors and Taints for Argo CD workloads
+	NodePlacement *ArgoCDNodePlacementSpec `json:"nodePlacement,omitempty"`
+
+	// Notifications defines whether the Argo CD Notifications controller should be installed.
+	Notifications ArgoCDNotifications `json:"notifications,omitempty"`
+
+	// Prometheus defines the Prometheus server options for ArgoCD.
+	Prometheus ArgoCDPrometheusSpec `json:"prometheus,omitempty"`
+
+	// RBAC defines the RBAC configuration for Argo CD.
+	RBAC ArgoCDRBACSpec `json:"rbac,omitempty"`
+
+	// Redis defines the Redis server options for ArgoCD.
+	Redis ArgoCDRedisSpec `json:"redis,omitempty"`
+
+	// Repo defines the repo server options for Argo CD.
+	Repo ArgoCDRepoSpec `json:"repo,omitempty"`
+
+	// RepositoryCredentials are the Git pull credentials to configure Argo CD with upon creation of the cluster.
+	RepositoryCredentials string `json:"repositoryCredentials,omitempty"`
+
+	// ResourceCustomizations customizes resource behavior. Keys are in the form: group/Kind. Please note that this is being deprecated in favor of ResourceHealthChecks, ResourceIgnoreDifferences, and ResourceActions.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Resource Customizations'",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text","urn:alm:descriptor:com.tectonic.ui:advanced"}
+	ResourceCustomizations string `json:"resourceCustomizations,omitempty"`
```

Comment:
I think we want to get rid of this, as mentioned in the earlier comment

Reply:
Will remove it in a separate PR.